conformance/lang/expressions/transactional-expr/transactional_expr.balt

```diff
@@ -0,0 +1,558 @@
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, check, optional-type, error, int, transaction-stmt, rollback-stmt, commit-action,
+        transaction:onRollback, var, explicit-anonymous-function-expr, isolated-function, transaction:Info, is-expr,
+        if-else-stmt
+
+function init() returns error? {
+    var onRollbackFunc = isolated function(transaction:Info? info, error? cause, boolean willTry) {
+        io:println(transactional);
+    };
+
+    transaction {
+        int? a = 10;
+
+        if a is int {
+            io:println(transactional); // output true
+            transaction:onRollback(onRollbackFunc); // output true
+            io:println(transactional); // output true
+            rollback;
+            io:println(transactional); // output false
+        } else {
+            check commit;
+        }
+
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, check, optional-type, error, int, transaction-stmt, commit-action, transactional-worker
+
+function init() returns error? {
+    transaction {
+        f1();
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+transactional function f1() {
+    transactional worker A {
+        io:println(transactional); // output true
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, module-class-defn, explicit-new-expr, method-call-expr, check, optional-type, error,
+        int, transaction-stmt, commit-action,
+
+class Class {
+    transactional function f1() {
+        io:println(transactional); // output true
+    }
+}
+
+function init() returns error? {
+    Class c = new;
+    transaction {
+        c.f1(); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, module-type-defn, object-type, object-constructor-expr, method-call-expr,
+        check, optional-type, error, int, transaction-stmt, commit-action
+
+type ClientObj client object {
+    transactional function f1();
+    transactional remote function f2();
+};
+
+function init() returns error? {
+    ClientObj c = client object {
+        transactional function f1() {
+            io:println(transactional);
+        }
+        transactional remote function f2() {
+            io:println(transactional);
+        }
+    };
+
+    transaction {
+        c.f1(); // output true
+        io:println(transactional); // output true
+        c->f2(); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
```

Comment:
Shall we run the PR against https://github.com/ballerina-platform/ballerina-lang/pull/35308 and validate the labels

Reply:
Have already validated the labels :)